### ISSUE TITLE
Accessibility Issues with Fluent Store for Screen Readers

### ISSUE THREAD

**Issue by MesterPerfect** [yoshiask/FluentStore#82]
### Before you start...

- [X] Have you checked if similar issue have already been reported?

### What windows are you seeing the problem on?

Windows 10 Release Preview

### What OS/App version/architecture are you seeing the problem on?

The issue appears with all applications on Windows 10 Version 22H2 (OS Build 19045.4717).

### Description

There are significant accessibility issues with Fluent Store when using screen readers, such as NVDA. For instance, in the list of applications, I am unable to read the name of any application. Instead of the application name, the text "FluentStore.ViewModels.AppViewModel 2 of 37" is displayed. Additionally, I cannot access the buttons to install applications, among other issues.

### Steps To Reproduce

1. Open the Fluent Store application.
2. Navigate to the 'Applications' tab.
3. Attempt to browse the applications.

### Expected behavior

I expect to be able to read the names of applications in the list and access the install buttons using screen readers like NVDA.

**Comment by MesterPerfect** [comment]
This is a comparison of NVDA's behavior with the Fluent Store and NVDA's behavior with the Microsoft Store.

https://github.com/user-attachments/assets/147ed0fe-dcb9-489d-860b-3185c923b7d3

**Comment by yoshiask** [comment]
Howdy! Thanks for taking the time to report these issues. I've introduced some changes in 0.4.0 that should improve the screen reader situation, including the `AppViewModel` bug you pointed out.

Regarding the install buttons, I did notice it takes several tabs to get to the button, but it is technically accessible (I'll still try to improve this, don't worry). Are you unable to select the Install button even after 4 or 5 tabs at all? Are you referring to the secondary options? If so, does hitting `Alt + Down` while the Install button is highlighted open the menu?

Thanks!

**Comment by MesterPerfect** [comment]
> Howdy! Thanks for taking the time to report these issues. I've introduced some changes in 0.4.0 that should improve the screen reader situation, including the `AppViewModel` bug you pointed out.
> 
> Regarding the install buttons, I did notice it takes several tabs to get to the button, but it is technically accessible (I'll still try to improve this, don't worry). Are you unable to select the Install button at all, even after 4 or 5 tabs? Are you referring to the secondary options? If so, does hitting `Alt + Down` while the Install button is highlighted open the menu?
> 
> Thanks!


Thank you for the quick response, and I apologize for the delay in getting back to you.

When I try to access the Collections tab and then use the Tab key to navigate to any element, I receive this message:

![image](https://github.com/user-attachments/assets/1e938617-db9d-4d09-b38d-de9f422f8693)

Also, if I bypass the message, navigation between the elements becomes sluggish, and I'm unable to activate the install buttons and other related options.
